db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreDataManager.java

```diff
@@ -0,0 +1,348 @@
+package org.commonjava.indy.cassandra.data;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.commonjava.indy.audit.ChangeSummary;
+import org.commonjava.indy.data.IndyDataException;
+import org.commonjava.indy.data.StoreEventDispatcher;
+import org.commonjava.indy.db.common.AbstractStoreDataManager;
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.Group;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.StoreType;
+import org.commonjava.indy.model.core.io.IndyObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@ApplicationScoped
+@Alternative
+public class CassandraStoreDataManager extends AbstractStoreDataManager
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    CassandraStoreQuery storeQuery;
+
+    @Inject
+    IndyObjectMapper objectMapper;
+
+    @Override
+    protected StoreEventDispatcher getStoreEventDispatcher()
+    {
+        return null;
+    }
+
+    @Override
+    protected ArtifactStore getArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.getArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    protected ArtifactStore removeArtifactStoreInternal( StoreKey key )
+    {
+        DtxArtifactStore dtxArtifactStore = storeQuery.removeArtifactStore( key.getPackageType(), key.getType(), key.getName() );
+        return toArtifactStore( dtxArtifactStore );
+    }
+
+    @Override
+    public void clear( ChangeSummary summary ) throws IndyDataException
+    {
+
+    }
+
+    @Override
+    public Set<ArtifactStore> getAllArtifactStores()
```

Comment:
Looking through the code, it's kind of alarming how much this is used. It's going to have terrible performance because it has to hit all nodes...I wonder if we can start reducing our dependence on that somehow?

Reply:
I think DefaultArtifactStoreQuery will use this extensively. Just look at all the streaming methods in that class, which are used a lot IIRC. For example, we use it when clearing a NFC entry by StoreKey and name, if the store is not a group reference. We also use it in the generic proxy, when trying to resolve a requested URL to a pre-existing repository.

I think we need to try to refactor the ArtifactStoreQuery interface, producing more targeted methods to provide a more efficient way to access things we do commonly. CTL-Click on the method in DefaultArtifactStoreQuery (In IntelliJ IDEA) will show how much a method gets used...FYI